pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -140,7 +150,10 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 
 	if !HasFinalizer(cd, hivev1.FinalizerDeprovision) {
 		cdLog.Debugf("adding clusterdeployment finalizer")
-		return reconcile.Result{}, r.addClusterDeploymentFinalizer(cd)
+		err = r.addClusterDeploymentFinalizer(cd)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
Does this change look ok to you @csrwng ? Fixed a problem with the unit tests.

Reply:
So the difference now is that adding the finalizer successfully will result in the sync loop continuing rather than stopping right here. I don't necessarily see a problem with it, but traditionally we've only done one update to the resource we're reconciling per sync loop invocation.

I think we should fix unit tests though because they seem mighty flaky. I am currently running tests on the master branch and they only succeed 3/5 times.

Reply:
Found the problem, back to original form.